### ISSUE TITLE
Errors from validation should pick an appropriate range for the error location

### ISSUE THREAD

**Issue by StephenWeatherford** [microsoft/vscode-azurearmtools#578]
We're only given the position, so currently the range length is 0.
Also, they should point to the parameter file when appropriate.

# Scenarios
## Incorrect parameter value type (same whether in parameter file or in defaultValue in template file)
ACTUAL:
![image](https://user-images.githubusercontent.com/6913354/81871048-f123a980-952b-11ea-8abf-bd6c6e983927.png)
EXPECTED:
![image](https://user-images.githubusercontent.com/6913354/81871463-b0786000-952c-11ea-9d43-38cff8ae1945.png)

**Comment by StephenWeatherford** [comment]
cc @bmoore-msft @neilpeterson From our Teams conversation.

**Comment by StephenWeatherford** [comment]
saw/params.sync.errors.part2.adjusterrors

**Comment by StephenWeatherford** [comment]
Does this still repro?

**Comment by bmoore-msft** [comment]
![image](https://user-images.githubusercontent.com/10090014/105393453-fae4cd00-5be1-11eb-8724-53ac3cc847a7.png)

Seems so...

**Comment by StephenWeatherford** [comment]
Here's another example:

```
{
    // Verify correct scope of expressions, variables, parameters

    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "resources": [
        {
            "name": "linkedDeployment2",
            "type": "Microsoft.Resources/deployments",
            "apiVersion": "2020-10-01",
            "properties": {
                "mode": "Incremental",
                "templateLink": {
                    "relativePath": "child.json"
                },
                "parameters": {
                    "childIntParam": {
                        // ERROR: This index causes an out of bounds array access in the child
                        "value": 1
                    }
                }
            }
        }
    ],
    "outputs": {}
}
```
child:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "childIntParam": {
            "type": "int"
        }
    },
    "variables": {
        "arrayVar": [
            "hello"
        ]
    },
    "resources": [
        {
            "type": "Microsoft.Insights/autoscalesettings",
            "apiVersion": "2014-04-01",
            "location": "East US",
            "tags": {
                "[concat('hidden-link:', resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]": "Resource",
                "displayName": "appInsights1 AutoScale Settings for appServicePlan1"
            },
            "dependsOn": [
                "[resourceId('Microsoft.Web/serverfarms', 'appServicePlan1')]"
            ],
            "properties": {
                "name": "appInsights1-appServicePlan1-AutoScaleSettings",
                "profiles": [
                    {
                        "name": "Default",
                        "capacity": {
                            "minimum": 1,
                            "maximum": 2,
                            "default": 1
                        },
                        "rules": [
                            {
                                "metricTrigger": {
                                    "metricName": "CpuPercentage",
                                    "metricResourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]",
                                    "timeGrain": "PT1M",
                                    "statistic": "Average",
                                    "timeWindow": "PT10M",
                                    "timeAggregation": "Average",
                                    "operator": "GreaterThan",
                                    "threshold": 80.0
                                },
                                "scaleAction": {
                                    "direction": "Increase",
                                    "type": "ChangeCount",
                                    "value": 1,
                                    "cooldown": "PT10M"
                                }
                            },
                            {
                                "metricTrigger": {
                                    "metricName": "CpuPercentage",
                                    "metricResourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]",
                                    "timeGrain": "PT1M",
                                    "statistic": "Average",
                                    "timeWindow": "PT1H",
                                    "timeAggregation": "Average",
                                    "operator": "LessThan",
                                    "threshold": 60.0
                                },
                                "scaleAction": {
                                    "direction": "Decrease",
                                    "type": "ChangeCount",
                                    "value": 1,
                                    "cooldown": "PT1H"
                                }
                            }
                        ]
                    }
                ],
                "enabled": false,
                "targetResourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]"
            },
            
            // This is valid only for a childIntParam value of 0
            "name": "[variables('arrayVar')[parameters('childIntParam')]]"
        },
        {
            "name": "appInsights1-CPUHigh-appServicePlan1",
            "type": "Microsoft.Insights/alertrules",
            "apiVersion": "2016-03-01",
            "location": "East US",
            "tags": {
                "[concat('hidden-link:', resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]": "Resource",
                "displayName": "appInsights1 CPUHigh rule for appServicePlan1"
            },
            "dependsOn": [
                "[resourceId('Microsoft.Web/serverfarms', 'appServicePlan1')]"
            ],
            "properties": {
                "name": "appInsights1-CPUHigh-appServicePlan1",
                "description": "The average CPU is high across all the instances of appServicePlan1",
                "isEnabled": false,
                "condition": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
                    "dataSource": {
                        "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource",
                        "resourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]",
                        "metricName": "CpuPercentage"
                    },
                    "operator": "GreaterThan",
                    "threshold": 90,
                    "windowSize": "PT15M"
                },
                "action": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleEmailAction",
                    "sendToServiceOwners": true,
                    "customEmails": [
                    ]
                }
            }
        },
        {
            "name": "appInsights1-LongHttpQueue-appServicePlan1",
            "type": "Microsoft.Insights/alertrules",
            "apiVersion": "2016-03-01",
            "location": "East US",
            "tags": {
                "[concat('hidden-link:', resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]": "Resource",
                "displayName": "appInsights1 LongHttpQueue rule for appServicePlan1"
            },
            "dependsOn": [
                "[resourceId('Microsoft.Web/serverfarms', 'appServicePlan1')]"
            ],
            "properties": {
                "name": "appInsights1-LongHttpQueue-appServicePlan1",
                "description": "The HTTP queue for the instances of appServicePlan1 has a large number of pending requests.",
                "isEnabled": false,
                "condition": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
                    "dataSource": {
                        "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource",
                        "resourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlan1')]",
                        "metricName": "HttpQueueLength"
                    },
                    "operator": "GreaterThan",
                    "threshold": 100.0,
                    "windowSize": "PT5M"
                },
                "action": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleEmailAction",
                    "sendToServiceOwners": true,
                    "customEmails": [
                    ]
                }
            }
        },
        {
            "name": "appInsights1-ServerErrors-webApp1",
            "type": "Microsoft.Insights/alertrules",
            "apiVersion": "2016-03-01",
            "location": "East US",
            "tags": {
                "[concat('hidden-link:', resourceGroup().id, '/providers/Microsoft.Web/sites/webApp1')]": "Resource",
                "displayName": "appInsights1 ServerErrors rule for webApp1"
            },
            "dependsOn": [
                "[resourceId('Microsoft.Web/sites', 'webApp1')]"
            ],
            "properties": {
                "name": "appInsights1-ServerErrors-webApp1",
                "description": "webApp1 has some server errors, status code 5xx.",
                "isEnabled": false,
                "condition": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
                    "dataSource": {
                        "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource",
                        "resourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/sites/webApp1')]",
                        "metricName": "Http5xx"
                    },
                    "operator": "GreaterThan",
                    "threshold": 0.0,
                    "windowSize": "PT5M"
                },
                "action": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleEmailAction",
                    "sendToServiceOwners": true,
                    "customEmails": [
                    ]
                }
            }
        },
        {
            "name": "appInsights1-ForbiddenRequests-webApp1",
            "type": "Microsoft.Insights/alertrules",
            "apiVersion": "2016-03-01",
            "location": "East US",
            "tags": {
                "displayName": "appInsights1 ForbiddenRequests rule for webApp1"
            },
            "dependsOn": [
                "[resourceId('Microsoft.Web/sites', 'webApp1')]"
            ],
            "properties": {
                "name": "appInsights1-ForbiddenRequests-webApp1",
                "description": "webApp1 has some requests that are forbidden, status code 403.",
                "isEnabled": false,
                "condition": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
                    "dataSource": {
                        "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource",
                        "resourceUri": "[concat(resourceGroup().id, '/providers/Microsoft.Web/sites/webApp1')]",
                        "metricName": "Http403"
                    },
                    "operator": "GreaterThan",
                    "threshold": 0,
                    "windowSize": "PT5M"
                },
                "action": {
                    "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleEmailAction",
                    "sendToServiceOwners": true,
                    "customEmails": [
                    ]
                }
            }
        },
        {
            "name": "appInsights1",
            "type": "Microsoft.Insights/components",
            "apiVersion": "2015-05-01",
            "location": "Central US",
            "kind": "web",
            "tags": {
                "displayName": "appInsights1 Component for webApp1"
            },
            "dependsOn": [
                "[resourceId('Microsoft.Web/sites', 'webApp1')]"
            ],
            "properties": {
                "applicationId": "webApp1",
                "application_Type": "web"
            }
        }
    ]
}
```

**Comment by StephenWeatherford** [comment]
dev/stephwe/whole-line-errors

**Comment by StephenWeatherford** [comment]
Fixed by #1241
